### PR TITLE
Adjust construction of transaction options (breaking change)

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Allow one to create `TransactionOptions` with more than one option, using `TransactionOptions.withOptions()`.
 - Allow one to mutate the options, using `setWithHashSign()`, `setWithGuardian()`

Breaking change (from v11):
 - Remove static function `withTxHashSignOptions`.

### CODE REVIEW

File: src/networkParams.ts [hunk 1]
```diff
@@ -63,31 +59,57 @@ export class TransactionOptions {
...
-    static withDefaultOptions(): TransactionOptions {
+    static withDefaultOptions() {
         return new TransactionOptions(TRANSACTION_OPTIONS_DEFAULT);
     }
 
     /**
-     * Creates a TransactionsOptions object with the setting for hash signing
+     * Creates a TransactionOptions object from a set of options.
+     */
+    public static withOptions(options: {
```
Comment (andreibancioiu) [inline src/networkParams.ts:71]: Or should we name it `fromFlags()`, `withFlags()`?

File: src/networkParams.ts [hunk 2]
```diff
@@ -63,31 +59,57 @@ export class TransactionOptions {
...
+     * Returns true if the "hash sign" option is set.
+     */
+    isWithHashSign(): boolean {
+        return (this.value & TRANSACTION_OPTIONS_TX_HASH_SIGN) == TRANSACTION_OPTIONS_TX_HASH_SIGN;
+    }
+
+    /**
+     * Returns true if the "guarded transaction" option is set.
      */
-    static withTxHashSignOptions(): TransactionOptions {
```
Comment (andreibancioiu) [inline src/networkParams.ts:75]: Or should we leave it, but as deprecated?

File: src/networkParams.ts (hunk 1 above)
Comment (schimih) [reply src/networkParams.ts:71]: My opinion is that we should stay with "withOptions( )" given the  `TransactionOptions` name.
Does it make sense to keep the function `withOptions( )` and remove the "with" prefix from the options names?

File: src/networkParams.ts (hunk 2 above)
Comment (schimih) [reply src/networkParams.ts:75]: I would remove it.

**Review by schimih — COMMENTED** [review]
some opinions

**Review by schimih — APPROVED** [review]

File: src/networkParams.ts (hunk 1 above)
Comment (andreibancioiu) [reply src/networkParams.ts:71]: Indeed. Fixed.

File: src/networkParams.ts (hunk 2 above)
Comment (andreibancioiu) [reply src/networkParams.ts:75]: Removed.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Pull request merged**